src/Entities/Player.cpp

```diff
@@ -2636,3 +2636,101 @@ void cPlayer::FreezeInternal(const Vector3d & a_Location, bool a_ManuallyFrozen)
 	m_FlyingMaxSpeed = FlyingMaxpeed;
 	m_IsFlying = IsFlying;
 }
+
+
+
+
+float cPlayer::GetLiquidHeightPercent(NIBBLETYPE a_Meta)
+{
+	if (a_Meta >= 8)
+	{
+		a_Meta = 0;
+	}
+	return (float)(a_Meta + 1) / 9.0f;
+}
+
+
+
+
+
+bool cPlayer::IsInsideOfBlock(BLOCKTYPE a_Block)
+{
+	BLOCKTYPE Block = m_World->GetBlock(static_cast<int>(GetPosX()), static_cast<int>(m_Stance), static_cast<int>(GetPosZ()));
```

Comment:
Use `FloorC` instead of the `static_cast`s, to make it explicit in what direction the coords are truncated.